packages/backend-defaults/src/entrypoints/urlReader/lib/GithubUrlReader.ts

```diff
@@ -187,6 +191,34 @@ export class GithubUrlReader implements UrlReaderService {
     }
 
     const { filepath } = parseGitUrl(url);
+
+    // If it's a direct URL we use readUrl instead
+    if (!filepath?.match(/[*?]/)) {
+      try {
+        const data = await this.readUrl(url, options);
+
+        return {
+          files: [
+            {
+              url: url,
+              content: data.buffer,
+              lastModifiedAt: data.lastModifiedAt,
+            },
+          ],
+          etag: commitSha,
+        };
+      } catch (error) {
+        assertError(error);
+        if (error.name === 'NotFoundError') {
+          return {
+            files: [],
+            etag: commitSha,
+          };
+        }
```

Comment:
Yep seen from the point of view of the `search` method this really seems like the only sensible behavior tbh. If we didn't we'd have completely different behaviors for search and exact URLs.